src/common/node/fileSystem.ts

```diff
@@ -136,7 +137,25 @@ export class FileSystem {
     }
 
     public mkDir(p: string): Q.Promise<void> {
-        return Q.nfcall<void>(fs.mkdir, p);
+        return Q.nfcall<void>(this.fs.mkdir, p);
+    }
+
+    public stat(path: string): Q.Promise<fs.Stats> {
+        return Q.nfcall<fs.Stats>(this.fs.stat, path);
+    }
+
+    public directoryExists(directoryPath: string): Q.Promise<boolean> {
+        return this.stat(directoryPath).then(stats => {
+            return stats.isDirectory();
+        }).catch(reason => {
+            return reason.code === "ENOENT"
+                ? false
+                : Q.reject<boolean>(reason);
+        });
+    }
+
+    public rmdir(dirPath: string): Q.Promise<void> {
+        return Q.nfcall<void>(this.fs.rmdir, dirPath);
```

Comment:
+1 to @dlebu A generic method in the common folder should not have any assumptions on the usage. If you are making assumptions, then change the name of the method to indicate that - ex, "rmEmptyDir"